File: 2024-2/Proyecto DAO SoftPub Java/SoftPubPersonaTest/src/softpubdbpersonatest/SeccionBOTest.java
package softpubdbpersonatest;

import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.bo.SeccionBO;
import pe.edu.pucp.softpub.persona.model.Seccion;

/**
 *
 * Autor: andres
 */
public class SeccionBOTest {

    private static SeccionBO seccionBO;

    public static void testSeccionBO() {
        System.out.println("\ntestSeccionBO");
        seccionBO = new SeccionBO();
        testSeccionBOInsertarSecciones();
        testSeccionBOListarTodos();
        testSeccionBOModificarSecciones();
        testSeccionBOListarTodos();
        testSeccionBOObtenerPorId();
        testSeccionBOEliminarSecciones();
    }

    private static void testSeccionBOEliminarSecciones() {
        System.out.println("\ntestSeccionBOEliminarSecciones");
        int resultado = seccionBO.eliminar("INF");
        resultado = seccionBO.eliminar("MEC");
        resultado = seccionBO.eliminar("ELE");
        resultado = seccionBO.eliminar("TEL");
    }

    private static void testSeccionBOObtenerPorId() {
        System.out.println("\ntestSeccionBOObtenerPorId");
        Seccion seccion = seccionBO.obtenerPorId("MEC");
        if (seccion != null) {
            System.out.print(seccion.getIdSeccion());
            System.out.print(", ");
            System.out.print(seccion.getSeccion());
            System.out.print(", ");
            System.out.println(seccion.getDepartamento());
        } else {
            System.out.println("No existe la seccion MEC\n");
        }

        seccion = seccionBO.obtenerPorId("TEL");
        if (seccion != null) {
            System.out.print(seccion.getIdSeccion());
            System.out.print(", ");
            System.out.print(seccion.getSeccion());
            System.out.print(", ");
            System.out.println(seccion.getDepartamento());
        } else {
            System.out.println("No existe la seccion TEL");
        }
    }

    private static void testSeccionBOListarTodos() {
        System.out.println("\ntestSeccionBOListarTodos");
        ArrayList<Seccion> listaSeccion = seccionBO.listarTodos();
        for (Seccion seccion : listaSeccion) {
            System.out.print(seccion.getIdSeccion());
            System.out.print(", ");
            System.out.print(seccion.getSeccion());
            System.out.print(", ");
            System.out.println(seccion.getDepartamento());
        }
    }

    private static void testSeccionBOModificarSecciones() {
        System.out.println("\ntestSeccionBOModificarSecciones");
        int resultado = seccionBO.modificar("INF", "INFORMÁTICA", "Ingeniería");
        resultado = seccionBO.modificar("TEL", "TELECO", "Ingeniería");

    }

    private static void testSeccionBOInsertarSecciones() {
        System.out.println("\ntestSeccionBOInsertarSecciones");
        int resultado = seccionBO.insertar("INF", "Ingeniería Informática", "Ingeniería");
        resultado = seccionBO.insertar("TEL", "Ingeniería de las Telecomunicaciones", "Ingeniería");
        resultado = seccionBO.insertar("ELE", "Ingeniería Electrónica", "Ingeniería");
        resultado = seccionBO.insertar("MEC", "Ingeniería Mecánica", "Ingeniería");
    }

}

File: 2024-2/0684/unidad07.b - Programación distribuida con BD/SoftPub_Front-end/SoftPubPersonaTest/SeccionBOTest.cs
using System;
using System.ComponentModel;
using SoftPubBO.ServicioWeb;
using SoftPubPersonaBO;

namespace SoftPubPersonaTest
{
    internal class SeccionBOTest
    {
        private static SeccionBO seccionBO;

        internal static void testSeccionBO()
        {
            System.Console.WriteLine("testSeccionBO");
            seccionBO = new SeccionBO();
            testSeccionBOInsertarSecciones();
            testSeccionBOListarTodos();
            testSeccionBOModificarSecciones();
            testSeccionBOListarTodos();
            testSeccionBOObtenerPorId();
            testSeccionBOEliminarSecciones();
            System.Console.ReadKey();
        }

        private static void testSeccionBOEliminarSecciones()
        {
            System.Console.WriteLine("testSeccionBOEliminarSecciones");
            int resultado = seccionBO.eliminar("INF");
            resultado = seccionBO.eliminar("MEC");
            resultado = seccionBO.eliminar("ELE");
            resultado = seccionBO.eliminar("TEL");
        }

        private static void testSeccionBOObtenerPorId()
        {
            System.Console.WriteLine("testSeccionBOObtenerPorId");
            seccion _seccion = seccionBO.obtenerPorId("MEC");
            if (_seccion != null)
            {
                System.Console.Write(_seccion.idSeccion);
                System.Console.Write(", ");
                System.Console.Write(_seccion.seccion1);
                System.Console.Write(", ");
                System.Console.WriteLine(_seccion.departamento);
            }
            else
            {
                System.Console.WriteLine("No existe la seccion MEC\n");
            }

            _seccion = seccionBO.obtenerPorId("TEL");
            if (_seccion != null)
            {
                System.Console.Write(_seccion.idSeccion);
                System.Console.Write(", ");
                System.Console.Write(_seccion.seccion1);
                System.Console.Write(", ");
                System.Console.WriteLine(_seccion.departamento);
            }
            else
            {
                System.Console.WriteLine("No existe la seccion TEL");
            }
        }

        private static void testSeccionBOListarTodos()
        {
            System.Console.WriteLine("testSeccionBOListarTodos");
            BindingList<seccion> listaSeccion = seccionBO.listarTodos();
            foreach (seccion _seccion in listaSeccion)
            {
                System.Console.Write(_seccion.idSeccion);
                System.Console.Write(", ");
                System.Console.Write(_seccion.seccion1);
                System.Console.Write(", ");
                System.Console.WriteLine(_seccion.departamento);
            }
        }

        private static void testSeccionBOModificarSecciones()
        {
            System.Console.WriteLine("testSeccionBOModificarSecciones");
            int resultado = seccionBO.modificar("INF", "INFORMÁTICA", "Ingeniería");
            resultado = seccionBO.modificar("TEL", "TELECO", "Ingeniería");

        }

        private static void testSeccionBOInsertarSecciones()
        {
            System.Console.WriteLine("testSeccionBOInsertarSecciones");
            int resultado = seccionBO.insertar("INF", "Ingeniería Informática", "Ingeniería");
            resultado = seccionBO.insertar("TEL", "Ingeniería de las Telecomunicaciones", "Ingeniería");
            resultado = seccionBO.insertar("ELE", "Ingeniería Electrónica", "Ingeniería");
            resultado = seccionBO.insertar("MEC", "Ingeniería Mecánica", "Ingeniería");
        }
    }
}